src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -340,6 +340,34 @@ public static function create_fragment( $html, $context = '<body>', $encoding =
 		return $processor;
 	}
 
+	/**
+	 * Creates an HTML processor in the full parsing mode.
+	 *
+	 * It's likely that a fragment parser is more appropriate, unless sending an
+	 * entire HTML document from start to finish. Consider a fragment parser with
+	 * a context node of `<body>`.
+	 *
+	 * Since UTF-8 is the only currently-accepted charset, if working with a
+	 * document that isn't UTF-8, it's important to convert the document before
+	 * creating the processor: pass in the converted HTML.
+	 *
+	 * @param string      $html                    Input HTML document to process.
+	 * @param string|null $known_definite_encoding Optional. If provided, specifies the charset used
+	 *                                             in the input byte stream. Currently must be UTF-8.
+	 * @return static|null The created processor if successful, otherwise null.
+	 */
+	public static function create_full_parser( $html, $known_definite_encoding = 'UTF-8' ) {
+		if ( 'UTF-8' !== $known_definite_encoding ) {
+			return null;
+		}
+
+		$processor                             = new static( $html, self::CONSTRUCTOR_UNLOCK_CODE );
+		$processor->state->encoding            = $known_definite_encoding;
+		$processor->state->encoding_confidence = 'certain';
```

Comment:
right now the fragment parser _only_ supports `UTF-8`, which would be `certain` confidence. because that's inferred from the parent document, I don't feel like we need to add more there yet. it's still my intention _only_ to support UTF-8, as I would prefer we get in the habit of converting before processing.

thoughts?

Reply:
UTF-8 only support seems like the right choice at this time.

It's good that we set the encoding and encoding confidence in the fragment parser. Before that change it was resulting in different behavior for `<meta charset="anything">` tags between the full and fragment parsers. That's fixed by always setting `certain` encoding confidence.